asv_bench/benchmarks/hash_functions.py

```diff
@@ -0,0 +1,50 @@
+import pandas as pd
+import numpy as np
+
+
+class IsinWithArange:
+    params = [
+        [np.float64, np.int64, np.object],  #
+        [
+            1_000,
+            2_000,
+            8_000,
+        ],  # problem when quadratic behavior is triggered: [100_000, 10_000_000],
+        [-2, 0, 2],
+    ]
+    param_names = ["dtype", "M", "offset_factor"]
+
+    def setup(self, dtype, M, offset_factor):
+        offset = int(M * offset_factor)
+        np.random.seed(42)
+        self.s = pd.Series(np.random.randint(offset, M + offset, 10 ** 6)).astype(dtype)
+        self.values = np.arange(M).astype(dtype)
+
+    def time_isin(self, dtype, M, offset_factor):
+        self.s.isin(self.values)
+
+
+class GH28303Example:
```

Comment:
can you give this a descriptive name and then in a comment `# see GH#28303`

Reply:
Done.